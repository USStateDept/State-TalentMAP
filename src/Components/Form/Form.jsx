import PropTypes from 'prop-types';
import { EMPTY_FUNCTION } from '../../Constants/PropTypes';

const Form = ({ children, className, onFormSubmit, ...rest }) => (
  <form className={className} onSubmit={onFormSubmit} {...rest}>
    {children}
  </form>
);

Form.propTypes = {
  children: PropTypes.node.isRequired, // should be valid form children
  className: PropTypes.string,
  onFormSubmit: PropTypes.func,
};

Form.defaultProps = {
  className: '',
  onFormSubmit: EMPTY_FUNCTION,
};

export default Form;
