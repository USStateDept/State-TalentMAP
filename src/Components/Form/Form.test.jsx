import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import sinon from 'sinon';
import Form from './Form';

describe('FormComponent', () => {
  it('is defined', () => {
    const wrapper = shallow(
      <Form onFormSubmit={() => {}}>
        <span>child</span>
      </Form>,
    );
    expect(wrapper).toBeDefined();
  });

  it('accepts a className', () => {
    const className = 'test-class';
    const wrapper = shallow(
      <Form onFormSubmit={() => {}} className={className}>
        <span>child</span>
      </Form>,
    );
    expect(wrapper.find(`.${className}`)).toBeDefined();
  });

  it('calls onFormSubmit', () => {
    const spy = sinon.spy();
    const wrapper = shallow(
      <Form onFormSubmit={spy}>
        <span>child</span>
      </Form>,
    );
    wrapper.find('form').props().onSubmit();
    sinon.assert.calledOnce(spy);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <Form onFormSubmit={() => {}}>
        <span>child</span>
      </Form>,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
