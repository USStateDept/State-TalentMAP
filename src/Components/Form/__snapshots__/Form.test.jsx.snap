// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormComponent matches snapshot 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <span>
    child
  </span>
</form>
`;
